components/FollowButton.vue

```diff
@@ -46,14 +55,7 @@ const followActionText = computed( () => {
 })
 
 const onFollowClick = debounce(async () => {
```

Comment:
This is where you should check for contract existence and create if necessary.  You can use the text of the button and the icon to show this action is pending.